### DISCUSSION TITLE
Problem in EDF_DbContext.Create() - Unable to initialize database driver of type 'NULL'.

### DISCUSSION THREAD

**Discussion by FilliFilli** [Arkensor/EnfusionPersistenceFramework#45] [Q&A]
Hello Arkensor and everyone,

I'm working on a Arma AI with learning capabilities and i need to connect to a MySQL Database for that.
It's like my long time dream but i'm stuck by days, also having some programming experience but not professional.

I explain below, thanks in advance for any help =)

The problem is that at the moment i call:
`EDF_DbContext dbContext = EDF_DbContext.Create(connectionInfo);`
it returns me "Unable to initialize database driver of type 'NULL'."

Investigating furthermore, the problem is probably above during the constructor of
`new EDF_DbConnectionInfoDriverType(EDF_WebProxyDbDriver);`
cause seems that inside it uses a method EDF_ReflectionUtils.GetAttributeParent().

This is used to detect a superior class in the stack, and to do the mapping between ConnectionInfo and DbDriver which is what i'm trying to achieve.

In particular this function is not being able to find the "#CreateAttributes" word inside the stacktrace, and for that reason returns -1 as startRead. Cant print the stacktrace to console since its inside your mod files and i can't edit.

Further info:
- i'm scripting inside a file pertaining to another project in the editor
- my script is launched via a SCR_BaseGameMode -> OnWorldPostProcess simply with the code below in this thread
- the main script which is launched is attached as file here (needed to change extension on txt since .c files are not supported uploading)

Is the problem of wrong stacktrace related to where/how i am launching the script? should it be launched in another manner?
What am i getting wrong to make the DbContext creation work?

Really THANKS in advance for any help :)

The launching of InitializeDatabase() inside SCR_BaseGameMode -> OnWorldPostProcess:

`class GAIA_GameMode_Class: SCR_BaseGameMode 
{
	// user script
	// code here
	
	private ref GAIA_DatabaseInit DBInit;
	private ref GAIA_MyPersistentInfo PersInfo;
	private ref GAIA_MyPersistentInfoRepository PersInfoRep;
	
	GAIA_DatabaseInit GetDBInit()
	{
		return DBInit;
	}
	

	GAIA_MyPersistentInfo GetPersInfo()
	{
		return PersInfo;
	}
	
	GAIA_MyPersistentInfoRepository GetPersInfoRep()
	{
		return PersInfoRep;
	}

	override void OnWorldPostProcess(World world)
	{
		/* code here */
		
		// Initialize the database
        DBInit = new GAIA_DatabaseInit();
        DBInit.InitializeDatabase();`

The script:

[ConnectToDB.txt](https://github.com/user-attachments/files/16628452/ConnectToDB.txt)

**Comment by Arkensor** [comment]
Hello,

`EDF_WebProxyConnectionInfoBase` is a base classs as the name says and not expected to be called like this. See how MongoDB was connected to setup your own for MySQL if you need it. You do not register via calling new on the driver type, its an attribute that needs to be placed above the driver class. https://github.com/Arkensor/EnfusionDatabaseFramework/blob/armareforger/src/Scripts/Game/Drivers/WebProxy/EDF_MongoDbDriver.c
